<launch>
    <include file="$(find pose_prediction)/launch/rs_camera.launch" />

    <param name="/path_of_model" value="$(find pose_prediction)/segments.pt"/> 
    <node pkg="pose_prediction" type="yolo_seg.py" name="yolo_seg" >
        
    </node>
    <node pkg="pose_prediction" type="cloud_segment" name="cloud_segment" output="screen" />
    
    <!-- 启动位姿预测节点 -->
    <node pkg="pose_prediction" type="temp_prediction" name="temp_prediction" output="screen">
        <!-- 目标点云文件路径 -->
        <param name="target_cloud_path" value="/home/smartdrone/position/saved_cloud.pcd" />
        
        <!-- 配准参数 -->
        <param name="cube_size" value="0.1" />
        <param name="voxel_size" value="0.005" />
        <param name="normal_radius" value="0.02" />
        <param name="feature_radius" value="0.05" />
        <param name="sac_ia_max_distance" value="0.01" />
        <param name="sac_ia_max_iterations" value="1000" />
        <param name="icp_max_distance" value="0.005" />
        <param name="icp_max_iterations" value="100" />
        <param name="icp_transformation_epsilon" value="1e-8" />
    </node>
    
</launch> 